filament/backend/src/metal/MetalBuffer.mm

```diff
@@ -68,7 +68,7 @@
     // If the GPU buffer is not private, just perform the copy
     if (mBuffer.storageMode != MTLStorageModePrivate) {
         memcpy(static_cast<uint8_t*>(mBuffer.contents), src, size);
-#if !TARGET_OS_IOS || TARGET_OS_MACCATALYST
+#if TARGET_OS_MACCATALYST || TARGET_OS_OSX
```

Comment:
Doesn't `initializeSupportedGpuFamilies` need an update? When building for visionOS, is `TARGET_OS_IOS` defined (if not, `MetalContext::highestSupportedFamily::apple` will be missing some info)?

Reply:
And what about `MetalDriver::readPixels()`?

Reply:
Same question for `SamplerStateCreator::operator()`

Reply:
Will be fixed in https://github.com/shapr3d/filament/pull/146.

`initializeSupportedGpuFamilies` doesn't need updating, as `@available(..., *)` will return true for visionOS, and it runs fine. `MetalDriver::readPixels()` needs an update, it's in the PR. `SamplerStateCreator::operator()` checks for old iOS devices that do not support sampler comparisons, which is not necessary on visionOS.

Reply:
> it's in the PR
which one?